File: connectors/connector-jira/src/test/scala/com/taskadapter/connector/jira/JiraFactoryTest.scala
package com.taskadapter.connector.jira

import com.taskadapter.model.{AllFields, AssigneeLoginName, DueDate, EstimatedTime, Id, Key, ReporterLoginName}
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class JiraFactoryTest extends FunSpec with Matchers {
  describe("default fields") {
    it("have Assignee") {
      new JiraFactory().getDefaultFieldsForNewConfig should contain(AllFields.assigneeLoginName)
    }
    it("skip optional ones") {
      new JiraFactory().getDefaultFieldsForNewConfig should contain noneOf(AllFields.reporterLoginName,
        AllFields.estimatedTime,
        AllFields.dueDate,
        AllFields.id, AllFields.key)
    }
  }
}

File: connectors/connector-jira/src/test/java/com/taskadapter/connector/jira/JiraFactoryTest.java
package com.taskadapter.connector.jira;

import com.taskadapter.model.AllFields;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class JiraFactoryTest {
    @Test
    public void haveAssignee() {
        assertThat(new JiraFactory().getDefaultFieldsForNewConfig()).contains(AllFields.assigneeLoginName);
    }

    @Test
    public void skipOptionalOnes() {
        assertThat(new JiraFactory().getDefaultFieldsForNewConfig()).doesNotContain(AllFields.reporterLoginName,
                AllFields.estimatedTime,
                AllFields.dueDate,
                AllFields.id, AllFields.key);
    }
}
